gpiozero/spi_devices.py

```diff
@@ -72,10 +111,10 @@ class AnalogInputDevice(SPIDevice):
     .. _analog to digital converters: https://en.wikipedia.org/wiki/Analog-to-digital_converter
     """
 
-    def __init__(self, bits=None, **spi_args):
-        if bits is None:
-            raise InputDeviceError('you must specify the bit resolution of the device')
+    def __init__(self, bits, **spi_args):
         self._bits = bits
+        self._min_value = -(2 ** bits)
+        self._range = 2 ** (bits + 1) - 1
```

Comment:
I'm struggling to understand what you're doing here? For a non-differential ADC I'd expect a `_min_value` of `0`, and a `_range` of `2 ** bits` (e.g 0 to 1023); and for a differential ADC I'd expect a `_min_value` of `-(2 ** (bits - 1))` with the same range of `2 ** bits` (e.g. -512 to 511) ?
But you've obviously studied the data-sheets closer than me, so maybe I'm missing something?? :-S